action/protocol/poll/staking_command.go

```diff
@@ -57,18 +61,18 @@ func NewStakingCommand(
 }
 
 func (sc *stakingCommand) CreateGenesisStates(ctx context.Context, sm protocol.StateManager) error {
+	// if v1 exists, bootstrap from v1 only
 	if sc.stakingV1 != nil {
-		if err := sc.stakingV1.CreateGenesisStates(ctx, sm); err != nil {
-			return err
-		}
+		return sc.stakingV1.CreateGenesisStates(ctx, sm)
 	}
 
-	if sc.stakingV2 != nil {
-		if err := sc.stakingV2.CreateGenesisStates(ctx, sm); err != nil {
-			return err
-		}
+	cands, err := sc.stakingV2.ActiveCandidates(ctx)
+	if err != nil {
+		return err
 	}
-	return nil
+	bcCtx := protocol.MustGetBlockchainCtx(ctx)
+	cands = sc.filterAndSortCandidatesByVoteScore(cands, bcCtx.Tip.Timestamp)
```

Comment:
if this part is missing in current code, is it doable to move this into stakingV2.CreateGenesis(), and handle there?

and here still directly call stakingV2.CreateGenesis(), StakingCommand is just combining V1 and V2, and call their respective func, feel that would be better

Reply:
then we need define a stakingv2 in poll package. staking package shouldn't do this part of work.